### PR TITLE
[IE CLDNN] ITT counters

### PR DESCRIPTION
**Pull request by egerashc**

### CODE REVIEW

File: inference-engine/thirdparty/clDNN/src/gpu/device_info.cpp [hunk 1]
```diff
@@ -223,7 +223,7 @@ device_info_internal::device_info_internal(const cl::Device& device) {
 
     supports_subgroups_short = extensions.find("cl_intel_subgroups_short") != std::string::npos;
 
-    supports_imad = get_imad_support(device);
+    supports_imad = get_imad_support(device) || 1;
```
[the remark this answers is not in the document]
Comment (egerashc) [reply inference-engine/thirdparty/clDNN/src/gpu/device_info.cpp]: fixed

File: inference-engine/src/cldnn_engine/cldnn_executable_network.cpp [hunk 2]
```diff
@@ -134,6 +139,7 @@ InferenceEngine::Parameter CLDNNExecNetwork::GetMetric(const std::string &name)
 }
 
 RemoteContext::Ptr CLDNNExecNetwork::GetContext() const {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "CLDNNExecNetwork::GetContext");
```
Comment (myshevts) [inline inference-engine/src/cldnn_engine/cldnn_executable_network.cpp:142]: Not sure for which specific task you are instrumneting the clDNN, but wrapping EVERY function (especially setters/getters that may be called many many times) seems redundant. At least for the perf debugging. Myabe for conditional compilation this is needed...

Comment (egerashc) [reply inference-engine/src/cldnn_engine/cldnn_executable_network.cpp]: fixed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 3]
```diff
@@ -299,6 +314,7 @@ void Program::InitProfileInfo(const std::string& layerName,
                               const std::string& layerType,
                               bool isCPU,
                               InferenceEngine::InferenceEngineProfileInfo::LayerStatus status, std::string parentId) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::InitProfileInfo");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:317]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 4]
```diff
@@ -282,13 +295,15 @@ void Program::AddPrimitiveToProfiler(const std::shared_ptr<ngraph::Node>& op,
...
                                      cldnn::primitive_id customOutputId) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::AddPrimitiveToProfiler_");
     primitivesToIRLayersMap[id] = { op->get_friendly_name() };
     primitiveIDs[id] = customOutputId.empty() ? id : customOutputId;
     profilingIDs.push_back(id);
 }
 
 void Program::AddInnerPrimitiveToProfiler(cldnn::primitive_id id, cldnn::primitive_id parentId,
                                           const std::shared_ptr<ngraph::Node>& op) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::AddPrimitiveToProfiler__");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:306]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 5]
```diff
@@ -282,13 +295,15 @@ void Program::AddPrimitiveToProfiler(const std::shared_ptr<ngraph::Node>& op,
 
 void Program::AddPrimitiveToProfiler(cldnn::primitive_id id, const std::shared_ptr<ngraph::Node>& op,
                                      cldnn::primitive_id customOutputId) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::AddPrimitiveToProfiler_");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:298]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 6]
```diff
@@ -274,6 +286,7 @@ std::vector<cldnn::primitive_id> Program::GetInputPrimitiveIDs(const std::shared
 
 void Program::AddPrimitiveToProfiler(const std::shared_ptr<ngraph::Node>& op,
                                      cldnn::primitive_id customOutputId) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::AddPrimitiveToProfiler");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:289]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 7]
```diff
@@ -248,6 +259,7 @@ void Program::CreateSingleLayerPrimitive(cldnn::topology& topology, const std::s
 }
 
 std::vector<cldnn::primitive_id> Program::GetInputPrimitiveIDs(const std::shared_ptr<ngraph::Node>& op) const {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::GetInputPrimitiveIDs");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:262]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 8]
```diff
@@ -154,19 +157,22 @@ int Program::GetMaxBatchSizeForSingleProgram() {
...
 
 void Program::PrepareBuild(InferenceEngine::InputsDataMap networkInputs, InferenceEngine::OutputsDataMap networkOutputs) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::PrepareBuild");
     m_topology.reset(new cldnn::topology());
     m_networkInputs = networkInputs;
     m_networkOutputs = networkOutputs;
 }
 
 void Program::CleanupBuild() {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::CleanupBuild");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:175]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 9]
```diff
@@ -154,19 +157,22 @@ int Program::GetMaxBatchSizeForSingleProgram() {
...
 std::shared_ptr<cldnn::program> Program::GetCompiledProgram(int program_id) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::GetCompiledProgram");
     if (program_id >= m_programs.size())
         THROW_IE_EXCEPTION << "Invalid program ID";
 
     return m_programs[program_id];
 }
 
 void Program::PrepareBuild(InferenceEngine::InputsDataMap networkInputs, InferenceEngine::OutputsDataMap networkOutputs) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::PrepareBuild");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:168]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 10]
```diff
@@ -154,19 +157,22 @@ int Program::GetMaxBatchSizeForSingleProgram() {
 }
 
 std::shared_ptr<cldnn::program> Program::GetCompiledProgram(int program_id) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::GetCompiledProgram");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:160]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 11]
```diff
@@ -136,6 +138,7 @@ Program::Program(InferenceEngine::CNNNetwork& network, std::shared_ptr<const cld
 }
 
 int Program::GetMaxBatchSizeForSingleProgram() {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::GetMaxBatchSizeForSingleProgram");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:141]: Not needed

File: inference-engine/src/cldnn_engine/cldnn_program.cpp [hunk 12]
```diff
@@ -41,6 +42,7 @@ void Program::ChangeInputBatch(int batch) {
 }
 
 void Program::ValidateInputs(const std::shared_ptr<ngraph::Node>& op, std::vector<size_t> validInputsCount) {
+    OV_ITT_SCOPED_TASK(itt::domains::CLDNNPlugin, "Program::ValidateInputs");
```
Comment (vladimir-paramuzov) [inline inference-engine/src/cldnn_engine/cldnn_program.cpp:45]: Not needed

**Review by vladimir-paramuzov — CHANGES_REQUESTED** [review]

**Review by vladimir-paramuzov — APPROVED** [review]

**Pull request merged**